datasource/src/main/scala/quasar/physical/s3/S3DatasourceModule.scala

```diff
@@ -41,11 +44,13 @@ import slamdata.Predef.{Stream => _, _}
 object S3DatasourceModule extends LightweightDatasourceModule {
   def kind: DatasourceType = s3.datasourceKind
 
-  def lightweightDatasource[F[_]: ConcurrentEffect: MonadResourceErr: Timer](config: Json)
+  def lightweightDatasource[F[_]: ConcurrentEffect: ContextShift: MonadResourceErr: Timer](config: Json)
       : F[InitializationError[Json] \/ Disposable[F, Datasource[F, Stream[F, ?], ResourcePath]]] = {
+    val ec = ExecutionContext.Implicits.global
```

Comment:
Is there any way I can get an `ExecutionContext` out of the implicit `ContextShift`? cc @djspiewak

Reply:
No. If you *need* an `ExecutionContext`, then what you can do is pass the implicit `ExecutionContext` around and just rely on the fact that `ContextShift` will implicitly derive from there. It's definitely suboptimal, but way better than fishing out global (which isn't even the correct pool).

Why do you need an `ExecutionContext` specifically?

Reply:
The new version of `http4s` (the one supporting fs2 1.0) uses `BlazeClientBuilder` to construct the HTTP client. The builder in turn [requires an `ExecutionContext`](https://github.com/http4s/http4s/blob/v0.19.0/blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala#L172) to be passed in. 

The previous version of `http4s` [uses the global `ExecutionContext` by default](https://github.com/http4s/http4s/blob/v0.18.13/blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientConfig.scala#L83). I could require an implicit `ExecutionContext` in this method, but unfortunately that changes the signature of `lightweightDatasource`, so it's no longer a loadable `DatasourceModule` .